src/paraswap.ts

```diff
@@ -427,25 +427,24 @@ export class ParaSwap {
     userAddress: Address,
     tokenAddress: Address,
     _provider?: any,
-    contractOptions?: ContractOptions,
+    sendOptions?: Omit<SendOptions, 'from'>,
   ): Promise<string> {
     return new Promise(async (resolve, reject) => {
       const spender = await this.getSpender();
```

Comment:
Since we are here, here's an unsolicited _by-the-way_ architectural observation
Guess what happens if `await this.getSpender()` throws
No, the wrapping promise won't reject. Nothing will happen, the promise will forever stay pending

Reply:
Or if contract creation breaks for that matter.

Reply:
I took me some time to see that we were inside `new Promise`, agree